cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDTransactions.java

```diff
@@ -0,0 +1,139 @@
+    @Given("I login")
+    public void login() {
+        navigator.navigateTo(Navigator.PGNAME_LOGIN);
+        LoginPage loginPage = new LoginPage(driver);
+        User user = new User(UserType.PERSISTANT, "joseph.sam@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    /**
+     * General Scenario Definitions
+     */
+    @Given("I am on the Transactions page")
+    public void iAmOnTheTranactionsPage() {
+        this.transactions.clickTab();
+        String expectedUrl = "http://localhost:5173/dashboard/transactions";
+        String actualUrl = this.transactions.getCurrentUrl();
+        Assert.assertEquals(actualUrl, expectedUrl);
+    }
+
+    /**
+     * Create Transaction Scenario Definitions
+     */
+
+    @And("I fill in the {string}, {string}, {string}, and {string}")
+    public void andIFillInThe(String name, String account, String amount, String category) {
+        transactionContent[0] = this.transactions.setName(name);
+        transactionContent[1] = this.transactions.setAccount(account);
+        transactionContent[2] = this.transactions.setAmount(amount);
+        transactionContent[3] = this.transactions.setCategory(category);
+    }
+
+    @Then("I can see the new transaction in my list")
+    public void iCanSeeTheNewTransactionInMyList() {
+        String actualTransaction = this.transactions.verifyTransactionDetails();
+        System.out.println(actualTransaction);
+    }
+
+    /**
+     * Read Transaction Scenario Definitions
+     */
+    @Then("I can see the list of all my transactions")
+    public void iCanSeeTheListOfAllMyTransactions() {
+        this.transactions.printTransactionTable();
+        Assert.assertTrue(this.transactions.printTransactionTable());
+    }
+
+    /**
+     * Update Transaction Scenario Definitions
+     */
+
+    @And("I update the {string}, {string}, {string}, and {string}")
+    public void andIUpdateThe(String name, String account, String amount, String category) {
+        this.transactions.updateName(name);
+        this.transactions.updateAccount(account);
+        this.transactions.updateAmount(amount);
+        this.transactions.updateCategory(category);
+    }
+
+    /**
+     * Delete Transaction Scenario Definitions
+     */
+    @When("I click the Trash Icon button")
+    public void iClickTheTrashIconButton() {
+        this.transactions.clickDeleteBtn();
+    }
+
+    @Then("the transaction is not in the list")
+    public void theTransactionIsNotInTheList() {
+        this.transactions.confirmDeletion();
+        Assert.assertTrue(this.transactions.confirmDeletion());
+    }
+
+    /**
+     * Category Filter Scenario Definitions
+     */
+
+    @And("I click a {string} to filter based on")
+    public void iClickToFilterBasedOn(String category) {
+        this.transactions.selectACategory(category);
+    }
+
+    @Then("only transactions with that category should be visible")
+    public void onlyTransactionsWithThatCategoryShouldBeVisible() {
```

Comment:
Low impact: This Then step does not ensure that anything happens.
Suggestion: Remove or add an assertion of some sort.